tpm2/src/util/tpm2_kmyth_object.c

```diff
@@ -34,6 +40,17 @@ void tpm2_init_kmyth_object_sensitive(TPM2B_AUTH object_auth,
   // For data, the data buffer size cannot be zero - we must populate the
   // buffer with data to be sealed and set the size to its length in bytes.
   sensitiveArea->sensitive.data.size = object_dataSize;
+  // TODO: Calling memcpy with either source or dest addresses set to NULL
+  // is technically undefined behavior. However, object_data is expected
+if ( (object_dataSize == 0) || (object_data == NULL) )
+{
+   sensitiveArea->sensitive.data.size = 0;
+}
+else
+{
+    sensitiveArea->sensitive.data.size = object_dataSize;
+    memcpy(&sensitiveArea.sensitive.data.buffer, object_data, sensitiveArea.sensitive.data.size);
+}
   memcpy(&sensitiveArea->sensitive.data.buffer, object_data,
          sensitiveArea->sensitive.data.size);
   if (object_dataSize > 0)
```

Comment:
This debug log statement will happen if object_dataSize > 0 but object_data == NULL. If we want to keep it, we should probably either just move the kmyth_log() statement into the else block above or use sensitiveArea->sensitive.data.buffer instead of object_dataSize in the if condition.

Reply:
Yep, moved it into the else block.